lib/config/configitem.cpp

```diff
@@ -415,22 +424,29 @@ bool ConfigItem::CommitNewItems(const ActivationContext::Ptr& context, WorkQueue
 			if (item->m_Abstract || item->m_Object)
 				continue;
 
-			items.emplace_back(item, true);
+			itemsByType[item->m_Type.get()].emplace_back(item, true);
+			++total;
 		}
 
 		m_UnnamedItems.swap(newUnnamedItems);
 	}
 
-	if (items.empty())
+	if (!total)
 		return true;
 
 	// Shuffle all items to evenly distribute them over the threads of the workqueue. This increases perfomance
 	// noticably in environments with lots of objects and available threads.
-	std::shuffle(std::begin(items), std::end(items), std::default_random_engine {});
+	for (auto& kv : itemsByType) {
+		upq.Enqueue([&kv]() {
+			std::shuffle(std::begin(kv.second), std::end(kv.second), std::default_random_engine{});
+		});
+	}
+
+	upq.Join();
```

Comment:
I'm afraid there are still `if (...) return;`s, so I'd rather keep this.

Reply:
Still, I wouldn't parallelize this. `CommitNewItems()` is called repeatedly, so even if there was some call where there are enough elements in these vectors that parallelizing this would speed up things a bit, this may easily be negated by the overhead in other calls.

I'd say unless you've explicitly identified the existing call as a bottleneck, there's no reason for parallelizing this. If shuffling $N$ elements in a single thread is fine, shuffling a total of $N = n_1 + ... + n_k$ items in $k$ separate calls is fine as well.